adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -2164,10 +2164,9 @@ trait MultisampleGenomicRDD[T, U <: MultisampleGenomicRDD[T, U]] extends Genomic
 }
 
 /**
- * A trait describing a GenomicRDD that also supports the Spark SQL APIs.
+ * A trait describing a GenomicDataset that also supports the Spark SQL APIs.
  */
-trait GenomicDataset[T, U <: Product, V <: GenomicDataset[T, U, V]] extends GenomicRDD[T, V] {
-
+trait DatasetBoundGenomicDataset[T, U <: Product, V <: GenomicDataset[T, U, V]] extends GenomicRDD[T, V] {
```

Comment:
You shouldn't need to redefine `val dataset: Dataset[U]` in this trait, but otherwise looks good!

Reply:
Also, this should extend `GenomicDataset`, not `GenomicRDD`.